src/supermarket/Gemfile

```diff
@@ -50,7 +50,7 @@ gem "unicorn"
 gem "unicorn-rails"
 gem "utf8-cleaner"
 gem "validate_url"
-gem "virtus", "1.0.2", require: false # Pin to a version before the handling of nil in collection coercion was fixed.
+gem "virtus", "2.0.0", require: false # Pin to a version before the handling of nil in collection coercion was fixed.
```

Comment:
sounds like from this comment we potentially can't upgrade here. It might be worth digging into the git history to see why that pin was made

Reply:
Yes @tas50 this Gem is discontinued: https://github.com/solnic/virtus and is not worth the upgrade. The maintainer of this gem has recommended to use some other Gems that are active on github e.g. https://github.com/dry-rb/dry-types
We should consider replacing it with the newer alternatives of this gem.
I'm keeping the Gem version same as current as upgrading to 1.0.3 itself is breaking the unit tests.